Sources/bitrise-step-swift-clone/Step.swift

```diff
@@ -0,0 +1,32 @@
+import Foundation
+import GitKit
+import TSCBasic
+
+struct SwiftCloneStep {
+    let config: Config
+
+    var gitCloneArgs: [String] {
+        var args = [config.repositoryURL, "--verbose"]
+        if let checkoutBranch = config.checkoutBranch {
+            args.append(contentsOf: ["--branch", checkoutBranch])
+        }
+        if let cloneDepth = config.cloneDepth {
+            args.append(contentsOf: ["--depth", cloneDepth])
+        }
+        return args
+    }
+
+    func createClonePath() throws -> AbsolutePath {
```

Comment:
is it more swifty to do -> var createClonePath: Result<AbsolutePath, Error>

Reply:
Maybe, yeah, I can try that. I honestly haven't sued the `Result` type all that much, but I think it is more "swifty" as you say.

Reply:
I tried implementing things this way, but didn't really like the results. I think using that `Result` type makes sense when you want to handle the `.failure()` case in a specific way, but in this case, we just want to bubble up and throw the error. So to me, just throwing it is a better choice. What do you think?